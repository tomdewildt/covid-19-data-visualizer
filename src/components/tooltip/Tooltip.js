import styled from "styled-components";
import PropTypes from "prop-types";

const Tooltip = styled.div`
    position: absolute; 
    top: ${ ( { y } ) => y }px;
    left: ${ ( { x } ) => x }px;
    min-width: 160px; 
    max-height: 240px;
    padding: ${ ( { theme } ) => theme.dimens.spacing[ 2 ] };
    border-radius: ${ ( { theme } ) => theme.dimens.borderRadius };
    background: ${ ( { theme } ) => theme.colors.black };
    color: ${ ( { theme } ) => theme.colors.white };
    z-index: 10;
`;
Tooltip.propTypes = {
    x: PropTypes.number.isRequired,
    y: PropTypes.number.isRequired,
};

const TooltipTitle = styled.h1`
    margin: ${ ( { theme } ) => `0 0 ${ theme.dimens.spacing[ 2 ] } 0` };
    font-size: ${ ( { theme } ) => theme.typography.fontSize };
`;

const TooltipText = styled.p`
    margin: 0;
    font-size: ${ ( { theme } ) => `calc(${ theme.typography.fontSize } * 0.75)` };
`;

Tooltip.Title = TooltipTitle;
Tooltip.Text = TooltipText;

export default Tooltip;
