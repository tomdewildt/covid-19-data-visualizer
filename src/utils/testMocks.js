/* eslint-disable react/prop-types */
import React from "react";

// Components
jest.mock( "@deck.gl/react", () => ( { children } ) => (
    <mock-deckgl>{ children }</mock-deckgl>
) );

// Assets
jest.mock( "../assets/gemeenten.json", () => ( [
    {
        type: "Feature",
        properties: { Gemeente: "Gemeente 1", Gemeentecode: 1, Coordinaten: [] },
        geometry: { type: "MultiPolygon", coordinates: [] },
    },
    {
        type: "Feature",
        properties: { Gemeente: "Gemeente 2", Gemeentecode: 2, Coordinaten: [] },
        geometry: { type: "MultiPolygon", coordinates: [] },
    },
    {
        type: "Feature",
        properties: { Gemeente: "Gemeente 3", Gemeentecode: 3, Coordinaten: [] },
        geometry: { type: "MultiPolygon", coordinates: [] },
    },
] ) );

