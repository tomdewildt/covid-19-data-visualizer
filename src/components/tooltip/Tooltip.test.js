import React from "react";
import { axe } from "jest-axe";

import { renderWithTheme } from "../../utils/testUtils";

import Tooltip from "./Tooltip";

describe( "Tooltip", () => {
    it( "renders consistently", () => {
        const { asFragment } = renderWithTheme( <Tooltip x={ 0 } y={ 0 } /> );
        const fragment = asFragment();

        expect( fragment ).toMatchSnapshot();
    } );

    it( "should have no axe violations", async () => {
        const { container } = renderWithTheme( <Tooltip x={ 0 } y={ 0 } /> );
        const result = await axe( container );

        expect( result ).toHaveNoViolations();
    } );

    it( "respects the x and y prop", () => {
        const { getByText } = renderWithTheme( <Tooltip x={ 50 } y={ 100 }>tooltip</Tooltip> );
        const tooltip = getByText( "tooltip" );

        expect( tooltip ).toHaveStyleRule( "top", "100px" );
        expect( tooltip ).toHaveStyleRule( "left", "50px" );
    } );
} );

describe( "Tooltip.Title", () => {
    it( "renders consistently", () => {
        const { asFragment } = renderWithTheme( <Tooltip.Title>title</Tooltip.Title> );
        const fragment = asFragment();

        expect( fragment ).toMatchSnapshot();
    } );

    it( "should have no axe violations", async () => {
        const { container } = renderWithTheme( <Tooltip.Title>title</Tooltip.Title> );
        const result = await axe( container );

        expect( result ).toHaveNoViolations();
    } );
} );

describe( "Tooltip.Text", () => {
    it( "renders consistently", () => {
        const { asFragment } = renderWithTheme( <Tooltip.Text>text</Tooltip.Text> );
        const fragment = asFragment();

        expect( fragment ).toMatchSnapshot();
    } );

    it( "should have no axe violations", async () => {
        const { container } = renderWithTheme( <Tooltip.Text>text</Tooltip.Text> );
        const result = await axe( container );

        expect( result ).toHaveNoViolations();
    } );
} );
