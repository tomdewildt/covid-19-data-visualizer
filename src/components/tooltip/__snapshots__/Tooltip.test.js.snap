// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tooltip renders consistently 1`] = `
<DocumentFragment>
  .c0 {
  position: absolute;
  top: 0px;
  left: 0px;
  min-width: 160px;
  max-height: 240px;
  padding: 0.5rem;
  border-radius: 0.5rem;
  background: #191a1a;
  color: #ffffff;
  z-index: 10;
}

<div
    class="c0"
    x="0"
    y="0"
  />
</DocumentFragment>
`;

exports[`Tooltip.Text renders consistently 1`] = `
<DocumentFragment>
  .c0 {
  margin: 0;
  font-size: calc(1rem * 0.75);
}

<p
    class="c0"
  >
    text
  </p>
</DocumentFragment>
`;

exports[`Tooltip.Title renders consistently 1`] = `
<DocumentFragment>
  .c0 {
  margin: 0 0 0.5rem 0;
  font-size: 1rem;
}

<h1
    class="c0"
  >
    title
  </h1>
</DocumentFragment>
`;
